test/ecto/adapters/mysql_test.exs

```diff
@@ -66,6 +68,19 @@ defmodule Ecto.Adapters.MySQLTest do
     IO.iodata_to_binary SQL.delete(prefx, table, filter, returning)
   end
 
+  setup meta do
```

Comment:
We don't need those tests at the adapter level. The unit tests you added in migration_test is enough.

Reply:
Without this the adapters tests will be broken. Ex:
`test create table with prefix (Ecto.Adapters.PostgresTest)
     test/ecto/adapters/postgres_test.exs:737
     ** (RuntimeError) could not find migration runner process for #PID<0.572.0>
     code: [{:add, :category_0, references(:categories), []}]}
     stacktrace:
       (ecto) lib/ecto/migration/runner.ex:221: Ecto.Migration.Runner.runner/0
       (ecto) lib/ecto/migration/runner.ex:65: Ecto.Migration.Runner.repo_config/2
       (ecto) lib/ecto/migration.ex:837: Ecto.Migration.references/2
       test/ecto/adapters/postgres_test.exs:739: (test)
`

I have just copied code from the migration test to make them pass, but I have probably added unnecessary code there. How should I proceed?

Reply:
I see. Maybe we should change those tests to, instead of using `reference`, they build the `%Reference{}` struct directly?

Reply:
Sure, I can try that.

Reply:
Hi @josevalim, I have changed to used %Reference{} and all tests are passing. Can you please check if it's all ok?